airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalResponse.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.workers.temporal;
+
+import java.util.Objects;
+
+public class TemporalResponse<T> {
+
+  private final T output;
+  private final JobMetadata metadata;
+
+  public static <T> TemporalResponse<T> error(JobMetadata metadata) {
+    return new TemporalResponse<>(null, metadata);
+  }
+
+  public static <T> TemporalResponse<T> success(T output, JobMetadata metadata) {
+    return new TemporalResponse<>(output, metadata);
+  }
+
+  public TemporalResponse(final T output, final JobMetadata metadata) {
+    this.output = output;
+    this.metadata = metadata;
+  }
+
+  public boolean isSuccess() {
+    return metadata.isSucceeded();
+  }
+
+  public T getOutput() {
+    return output;
```

Comment:
Should that throw an IllegalStateException if it is not success?

Reply:
going to go with optional and javadoc comment. but i agree with the spirit fo the question.